### ISSUE TITLE
externals are mangled if a non-external with the same name is in the mangleMap

### ISSUE THREAD

**Issue by dekdevy** [CodyJasonBennett/shaderkit#8]
Shader 1:
```
void main() {
  vec3 light = vec3(1.0);
}
```

Shader 2:
```
out vec3 light;
void main() {
  light = vec3(1.0);
}
```

Even if the mangler is instructed to not mangle externals, the second shaders external `light` will be mangled.

**Comment by CodyJasonBennett** [comment]
Released [v0.1.12](https://github.com/CodyJasonBennett/shaderkit/releases/tag/v0.1.12) for clashing between shader externals, but haven't addressed swizzling or member expressions yet (needs scope tracking #10 or a proper AST #14).

**Comment by CodyJasonBennett** [comment]
Have a fix in [v0.1.13](https://github.com/CodyJasonBennett/shaderkit/releases/tag/v0.1.13).
